enhancements/kube-apiserver/stability-api-removal-notifications.md

```diff
@@ -0,0 +1,159 @@
+---
+title: api-removal-notifications
+authors:
+  - "@sanchezl"
+reviewers:
+  - TBD
+  - "@deads2k"
+approvers:
+  - TBD
+creation-date: 2021-02-01
+last-updated: 2021-02-01
+status: provisional
+---
+
+# API Removal Notifications
+
+> This is the title of the enhancement. Keep it simple and descriptive. A good title can help communicate what the enhancement is and should be considered as part of any review.
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+Notify customers that an API that will be removed in the next release is in use.
+
+## Motivation
+
+Kubernetes is being more aggressive with regards to the removal of APIs that have been deprecated or pre-release APIs that have reached a dead-end and will never reach GA. Additionally, the ability to re-enable a previously removed API has been removed.
+
+Customers should be notified that an API that will be removed in the next release is in use so that they can take action to avoid failure when upgrading to the next release. 
+
+### Goals
+
+1. Notify customers an API that will be removed in the next release is in use.
+2. Document how to identify the workload that is using the API to be removed.
+
+### Non-Goals
+
+1. Automatically identify the workload using the API to be removed.
+2. Prevent the customer from attempting to upgrade when an API that will be removed in the next release is in use.
+3. Notify customers that an upstream API is not supported on OpenShift.
+
+## Proposal
+
+Fire an `APIRemovedInNextReleaseInUse` alert when an API that will be removed in the the next release in in use.
+
+On EUS releases, also fire an `APIRemovedInNextEUSReleaseInUse` alert when an API that will be removed in the next extended support release is in use.
```

Comment:
what should be the behaviour of these. Will they be conditional in some sense? I.e. non-EUS customers won't see the second alert?

Reply:
> non-EUS customers won't see the second alert?
@sttts AFAIK there is no "EUS flag" available. So non-EUS customers on an EUS compatible release (4.6 or 4.10), would also see this alert.

Without an "EUS flag", the alert definition would be added/removed in the specific release branches.

Reply:
Has @sdodson 's EUS enhancement specified this "flag"?